src/tool/csv2yaml.cpp

```diff
@@ -4115,3 +4137,111 @@ static bool read_constdb(char* fields[], int columns, int current) {
+		baselv = (uint16)atoi(fields[3]);
+		joblv = (uint16)atoi(fields[4]);
+		offset = 5;
+	}
+	else if (columns == 3 + MAX_PC_SKILL_REQUIRE * 2) {
+		baselv = joblv = 0;
+		offset = 3;
+	}
+	else {
+		ShowWarning("pc_readdb_skilltree: Invalid number of colums in skill %hu of job %d's tree.\n", skill_id, class_);
+		return false;
+	}
+
+	const char* constant = constant_lookup(class_, "JOB_");
+	if (constant == nullptr) {
+		ShowWarning("pc_readdb_skilltree: Invalid job class %d specified.\n", class_);
+		return false;
+	}
+	std::string job_name(constant);
+
+	std::string* skill_name = util::umap_find( aegis_skillnames, skill_id );
+
+	if( skill_name == nullptr ){
+		ShowWarning("pc_readdb_skilltree: Unable to load skill %hu into job %d's tree.\n", skill_id, class_);
+		return false;
+	}
+
+	uint16 skill_lv = (uint16)atoi(fields[2]);
+
+	std::vector<s_skill_tree_entry_csv> *job = util::map_find(skill_tree, class_);
+	bool exists = job != nullptr;
+
+	s_skill_tree_entry_csv entry;
+
+	entry.skill_name = *skill_name;
+	entry.skill_id = skill_id;
+	entry.skill_lv = skill_lv;
+	entry.baselv = baselv;
+	entry.joblv = joblv;
+
+	for (uint16 i = 0; i < MAX_PC_SKILL_REQUIRE; i++) {
+		uint16 req_skill_id = (uint16)atoi(fields[i * 2 + offset]);
+		skill_lv = (uint16)atoi(fields[i * 2 + offset + 1]);
+
+		if (req_skill_id == 0)
+			continue;
+
+		skill_name = util::umap_find( aegis_skillnames, req_skill_id );
+
+		if( skill_name == nullptr ){
+			ShowWarning("pc_readdb_skilltree: Unable to load requirement skill %hu into job %d's tree.", req_skill_id, class_);
+			return false;
+		}
+		entry.need.insert({ *skill_name, skill_lv });
+	}
+
+	if (exists)
+		job->push_back(entry);
+	else {
+		std::vector<s_skill_tree_entry_csv> tree;
+
+		tree.push_back(entry);
+		skill_tree.insert({ class_, tree });
+	}
+
+	return true;
+}
+
+static bool pc_readdb_skilltree_yaml(void) {
+	for (const auto &it : skill_tree) {
+		body << YAML::BeginMap;
+		body << YAML::Key << "Job" << YAML::Value << constant_lookup(it.first, "JOB_");
+		body << YAML::Key << "Tree";
+		body << YAML::BeginSeq;
+		for (const auto &subit : it.second) {
+			body << YAML::BeginMap;
+			body << YAML::Key << "Name" << YAML::Value << subit.skill_name;
+			body << YAML::Key << "MaxLevel" << YAML::Value << subit.skill_lv;
+			if (!subit.need.empty()) {
+				body << YAML::Key << "Requires";
+				body << YAML::BeginSeq;
+				for (const auto &terit : subit.need) {
+					body << YAML::BeginMap;
+					body << YAML::Key << "Name" << YAML::Value << terit.first;
+					body << YAML::Key << "Level" << YAML::Value << terit.second;
+					body << YAML::EndMap;
+				}
+				body << YAML::EndSeq;
+			}
+			if (subit.baselv > 0)
+				body << YAML::Key << "BaseLevel" << YAML::Value << subit.baselv;
+			if (subit.joblv > 0)
+				body << YAML::Key << "JobLevel" << YAML::Value << subit.joblv;
+			body << YAML::EndMap;
+		}
+		body << YAML::EndSeq;
+		body << YAML::EndMap;
+		return true;
+	}
```

Comment:
```suggestion
	}
}
```

Missing a final curly bracket.